crates/teloxide-core/src/types/message.rs

```diff
@@ -1816,6 +1816,24 @@ impl Message {
     }
 }
 
+/// Implemented for syntax sugar, see issue <https://github.com/teloxide/teloxide/issues/1143>
+impl IntoIterator for Message {
+    type Item = Message;
+    type IntoIter = std::vec::IntoIter<Self::Item>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        vec![self].into_iter()
```

Comment:
simply `[self].into_iter()` should work as well, but without allocation

Reply:
Not exactly, `[self].into_iter()` will return `IntoIter<Message, 1>`, different from `IntoIter<Message>`

Reply:
Okay, nm then. Just haven't noticed the associated type :)